lib/web/desktop.go

```diff
@@ -103,17 +102,35 @@ func createDesktopConnection(
 	//
 	// In the future, we may want to do something smarter like latency-based
 	// routing.
-	service := winServices[rand.Intn(len(winServices))]
-	log = log.WithField("windows-service-uuid", service.GetName())
-	log = log.WithField("windows-service-addr", service.GetAddr())
-	serviceCon, err := site.DialTCP(reversetunnel.DialParams{
-		From:     &utils.NetAddr{AddrNetwork: "tcp", Addr: r.RemoteAddr},
-		To:       &utils.NetAddr{AddrNetwork: "tcp", Addr: service.GetAddr()},
-		ConnType: types.WindowsDesktopTunnel,
-		ServerID: service.GetName() + "." + ctx.parent.clusterName,
-	})
+	winDesktops, err := ctx.unsafeCachedAuthClient.GetWindowsDesktopsByName(r.Context(), desktopName)
+	var serviceCon net.Conn
+outer:
+	for _, desktop := range winDesktops {
```

Comment:
I would shuffle `winDesktops` first before looping through them. Otherwise, a bunch of attempts to connect to the same desktop may all go through the same Windows Desktop Service, rather than sharing the load equally.